[ci] 2025-04-21 spurious `hypothesis` failure

Starting today we are seeing failures like this even on trivial PRs. This needs investigation

One example:
https://github.com/single-cell-data/TileDB-SOMA/actions/runs/14580240642/job/40895167378

```
FAILED apis/python/tests/ht/test_ht_io_anndata.py::test_roundtrip_from_anndata_to_anndata - AssertionError: {'dictionary_item_removed': ["root['soma_']"]}
assert {'dictionary_...ot['soma_']"]} == {}
  
  Left contains 1 more item:
  {'dictionary_item_removed': ["root['soma_']"]}
  
  Full diff:
  - {}
  + {
  +     'dictionary_item_removed': ["root['soma_']"],
  + }
Falsifying example: test_roundtrip_from_anndata_to_anndata(
    tmp_path_factory=TempPathFactory(_given_basetemp=None,
     _trace=<pluggy._tracing.TagTracerSub object at 0x10e190190>,
     _basetemp=PosixPath('/private/var/folders/qn/7t0vq3ts721cmgt0tgrtgzl80000gn/T/pytest-of-runner/pytest-0'),
     _retention_count=3,
     _retention_policy='all'),
    data=data(...),
    adata=AnnData object with n_obs × n_vars = 55 × 41
        obs: 'obs_0', 'obs_1', 'obs_2'
        var: 'var_0', 'var_1', 'var_2', 'var_3', 'var_4', 'var_5', 'var_6', 'var_7'
        uns: '0', 'soma_',
    measurement_name='0',
    context=<tiledbsoma.options._soma_tiledb_context.SOMATileDBContext object at 0x10f207290>,
)
Draw 1: '0'
Draw 2: '00'

You can reproduce this example by temporarily adding @reproduce_failure('6.131.6', b'AXicczR31GR0ZHZkdGQAkhBs3mhAT8jgyABCDAyMjhyOTCguAWEQZHJkBZKajQaJjIMdMzBA/QP2kgYDKnCEQ1TECNQLFoXRCIoRixhYZWtxfm5iPJgNFms0aDIwAACBiWtQ') as a decorator on your test case
```

--- a/apis/python/tests/ht/test_ht_io_anndata.py
+++ b/apis/python/tests/ht/test_ht_io_anndata.py
@@ -27,7 +27,6 @@ import hypothesis as ht
 import numpy as np
 import numpy.typing as npt
 import pandas as pd
-import pytest
 import scipy.sparse as sp
 from hypothesis import given, settings
 from hypothesis import strategies as st
@@ -190,9 +189,10 @@ def dataframes(
 def keys() -> st.SearchStrategy[str]:
     # if sc-63410_workaround, we can't handle anything that isn't a legal posix path name.
     if HT_TEST_CONFIG["sc-63410_workaround"]:
-        return posix_filename()
+        stgy = posix_filename()
     else:
-        return st.text(string.printable, min_size=1, max_size=20)
+        stgy = st.text(string.printable, min_size=1, max_size=20)
+    return stgy.filter(lambda s: not s.startswith("soma_"))
 
 
 # AnnData <= 0.10 does not support scipy sparse_array
@@ -568,11 +568,6 @@ def assert_uns_equal(src_adata: anndata.AnnData, read_adata: anndata.Anndata) ->
     if diff.get("type_changes", None) == {}:
         del diff["type_changes"]
 
-    # Started 2025-04-21 even on trivial/unrelated PRs: needs investigation.
-    # assert diff == {}, repr(diff.to_dict())
-    if diff != {}:
-        pytest.xfail("https://github.com/single-cell-data/TileDB-SOMA/issues/4000")
-
 
 def assert_anndata_equal(
     src_adata: anndata.AnnData, read_adata: anndata.AnnData
@@ -604,10 +599,6 @@ def assert_anndata_equal(
     assert_uns_equal(src_adata, read_adata)
 
 
-@ht.reproduce_failure(
-    "6.131.6",
-    b"AXicczR31GR0ZHZkdGQAkhBs3mhAT8jgyABCDAyMjhyOTCguAWEQZHJkBZKajQaJjIMdMzBA/QP2kgYDKnCEQ1TECNQLFoXRCIoRixhYZWtxfm5iPJgNFms0aDIwAACBiWtQ",
-)
 @settings(
     suppress_health_check=(ht.HealthCheck.function_scoped_fixture,),
     deadline=timedelta(milliseconds=2500),

